teachertool/src/components/HomeScreen.tsx

```diff
@@ -112,11 +173,76 @@ const GetStarted: React.FC = () => {
     );
 };
 
+interface DataCarouselProps {
+    title: string;
+    cardsUrl: string;
+}
+
+const CardCarousel: React.FC<DataCarouselProps> = ({ title, cardsUrl }) => {
+    const [cardSet, setCardSet] = useState<CarouselCardSet | undefined>();
+    const [fetchStatus, setFetchStatus] = useState<RequestStatus | undefined>();
+
+    useJsonDocRequest(cardsUrl, setFetchStatus, setCardSet);
+
+    return (
+        <>
+            <div className={css.carouselRow}>
+                <div className={css.rowTitle}>
+                    <h2>{title}</h2>
+                </div>
+                {(fetchStatus === "loading" || fetchStatus === "error") && (
```

Comment:
Is there a plan to split this out in the future? Or maybe have retry logic? I feel like it would be a bit frustrating if you think that a card is going to load and it just never does.